cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueAdmin.java

```diff
@@ -180,12 +190,31 @@ public boolean doTruncateTable(@SuppressWarnings("unused") QueueName queueName)
     return column;
   }
 
+  // TODO: CDAP-1177 Move these functions to an abstract base class to share with LevelDBQueueAdmin
+  /**
+   * @param queueTableName actual queue table name
+   * @return namespace id that this queue belongs to
+   */
+  public static String getNamespaceId(String queueTableName) {
+    // last three parts are namespaceId (optional - in which case it will be the default namespace), appName and flow
+    String[] parts = queueTableName.split("\\.");
+    String namespaceId;
+    if (parts.length == 5) {
+      // cdap.system.queue.<app>.<flow>
+      namespaceId = Constants.DEFAULT_NAMESPACE;
+    } else {
```

Comment:
Do you want to make sure you check for length of 6 - otherwise if things change upstream this will cause a problem. It's better to be precise here.